definitions/procedures/service/list.rb

```diff
@@ -16,12 +16,22 @@ def run
 
     def unit_files_list(services)
       if systemd_installed?
-        regex = services.map { |service| "^#{service.name}.service" }.join('\|')
-        execute("systemctl list-unit-files | grep '#{regex}'")
+        execute("systemctl list-unit-files --type=service | \
+                 grep '#{build_regex_for_services(services)}'")
       else
         regex = services.map { |service| "^#{service.name} " }.join('\|')
         execute("chkconfig --list 2>&1 | grep '#{regex}'")
       end
     end
+
+    def build_regex_for_services(services)
+      services.map do |service|
+        if service.instance?
+          "^#{service.instance_parent_unit}.service"
+        else
+          "^#{service.name}.service"
+        end
```

Comment:
```suggestion
        if service.respond_to?(:instance_parent_unit) && service.instance_parent_unit
          "^#{service.instance_parent_unit}.service"
        else
          "^#{service.name}.service"
        end
```

I think no need of `instance?` method.